File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& price) {
        
    int maxPro = 0;
    int n = price.size();
    int minPrice = INT_MAX;

    for (int i = 0; i < price.size(); i++) {
        minPrice = min(minPrice, price[i]);
        maxPro = max(maxPro, price[i] - minPrice);
    }
    
    return maxPro;
}


        
    
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;
        int maxProfit = 0;

        for (int price : prices) {
            // update min price if current price is lower
            if (price < minPrice) {
                minPrice = price;
            } 
            // check if selling today gives better profit
            else {
                maxProfit = Math.max(maxProfit, price - minPrice);
            }
        }
        return maxProfit;
    }
}
